conbench/entities/benchmark_result.py

```diff
@@ -1128,33 +1087,58 @@ class _BenchmarkResultCreateSchema(marshmallow.Schema):
         metadata={
             "description": conbench.util.dedent_rejoin(
                 """
-                Identifier for a Run (required). This can be the ID of a known
-                Run (as returned by /api/runs) or a new ID in which case a new
-                Run entity is created in the database.
+                Unique identifier for a "run" of benchmarks (typically a single run of a
```

Comment:
- Can we make this property already optional now? I'd love that in the short term :). But yeah, can do later.
- "Unique identifier" might sound slightly misleading (like the database helps maintaining uniqueness; technically from the system's point of view it's an arbitrary identifier).

We should specify:

> the API does not ensure uniqueness (and, correspondingly, does not detect collisions). If your use case relies on this grouping construct then use a client-side ID generation scheme with negligible likelihood for collisions (e.g., UUID type 4 or similar).


Overall, let's maybe introduce this with

> Arbitrary identifier that you can use to group benchmark results.

Reply:
>Can we make this property already optional now?

Yeah I think I'll do this later because I'm afraid of wading through our test suite. :) you know the feeling

Good docs ideas!